tests/kernel/common/src/irq_offload.c

```diff
@@ -142,13 +142,20 @@ __no_optimization void test_nop(void)
 	 */
 	ztest_test_skip();
 #endif
-#elif defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE) || \
-	defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
-	/* do 4 nop instructions more to cost cycles */
-	arch_nop();
-	arch_nop();
-	arch_nop();
-	arch_nop();
+#elif defined(CONFIG_CPU_CORTEX_M)
+	/* The ARMv6-M Architecture manual states the following:
+	 * No Operation does nothing. This instruction can be used
+	 * for code alignment purposes.
+	 *
+	 * NOTE: The timing effects of including a NOP instruction
+	 * in code are not guaranteed. It can increase execution time,
+	 * leave it unchanged, or even reduce it. NOP instructions are
+	 * therefore not suitable for timing loops.
+	 *
+	 * A Similar statement can be found in the ARMv8-M Architecture Manual.
+	 * So we skip this test.
+	 */
+	ztest_test_skip();
```

Comment:
Since this is applicable to all ARM architecture variants (i.e. ARMv6/ARMv7/ARMv8, -A/-R/-M), we might as well just check for `defined(CONFIG_ARM) || defined(CONFIG_ARM64)`.

Also note that we already handle this for ARMv8-A below, so it would be good to expand on that.
https://github.com/zephyrproject-rtos/zephyr/blob/502ce2870fca78aa8da17d2cabf120230000ea4d/tests/kernel/common/src/irq_offload.c#L180-L181

Reply:
@stephanosio I have updated the implementation. I have reused earlier macros `CONFIG_ARMV6_M_ARMV8_M_BASELINE` and `CONFIG_ARMV7_M_ARMV8_M_MAINLINE` and avoided increasing the scope of the platforms covered as I have not investigated all.

Reply:
This is universal to all ARM architecture variants.

Looking at the ["ARM Architecture Reference Manual ARMv7-A and ARMv7-R edition"](https://developer.arm.com/documentation/ddi0406/cb/Application-Level-Architecture/Instruction-Details/Alphabetical-list-of-instructions/NOP):

> The timing effects of including a NOP instruction in a program are not guaranteed. It can increase execution time, leave it unchanged, or even reduce it. Therefore, NOP instructions are not suitable for timing loops.

For ARMv8-R, "ARM Architecture Reference Manual Supplement for the ARMv8-R AArch32 architecture profile" states the behaviour of `NOP` instruction is unchanged from that of the ARMv8-A, and the ARMv8-A ISA manual states the same thing as the above.

We can safely say that it is incorrect for this test to rely on the NOP timing delay effect on any ARM family architecture variants, so please update it as suggested earlier; otherwise, this test will still be technically broken for the currently supported ARMv7-A, ARMv7-R and ARMv8-R variants.

Reply:
I agreed with stephanosio that it is universal to all ARM archs. we can use `defined(CONFIG_ARM) || defined(CONFIG_ARM64)` for all ARM profiles. Or at least, `CONFIG_ARMV8_A ` can be replaced with `CONFIG_ARM64`